flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStKeyedStateBackend.java

```diff
@@ -147,15 +160,30 @@ public <SV, S extends State> S createState(@Nonnull StateDescriptor<SV> stateDes
     @Override
     @Nonnull
     public StateExecutor createStateExecutor() {
-        // TODO: Make io parallelism configurable
-        return new ForStStateExecutor(4, db, optionsContainer.getWriteOptions());
+        synchronized (lock) {
```

Comment:
Also shutdown in the close method ?
`close` is used to close resources created internally.
`dispose` is used to dispose resources including native resources.

Reply:
You means that `StateExecutor#shutdown `should be invoked in the `close` method, not the `dispose` method ?

Maybe we need introduce aonther `closed` flag in ForStKeyedStateBackend to prevent "createStateExecutor after close".

Reply:
I mean that it should be invoked in both `close` and `dispose` method which is similar to `RocksDBKeyedStateBackend`.
The semantic of `close` and `dispose` described above is also from it.
Otherwise the `close` method seems not so necessary.
WDYT?

Reply:
1. I think `StateExecutor#shutdown` should be invoked **exactly once**. (The `CloseableRegistry` in `RocksDBKeyedStateBackend` has the similar semantics).
2. As for the semantics of `close` and `dispose`,  StateBackend should call `close` method to close some dynamic threads and input/output streams first, then it can dispose some resources.

So I think the more reasonable way is to shutdown the stateExecutors in `close` method,   and force to close the `ForStKeyedStateBackend` in `dispose` method if the `ForStKeyedStateBackend` has not been closed yet.

WDYT?

Reply:
Thanks for the update.
It makes sense to me.